diff --git a/REE-Lib/OtherFiles/EFX/EfxBasics.cs b/REE-Lib/OtherFiles/EFX/EfxBasics.cs
index 159d632..0e9c563 100644
--- a/REE-Lib/OtherFiles/EFX/EfxBasics.cs
+++ b/REE-Lib/OtherFiles/EFX/EfxBasics.cs
@@ -480,7 +480,7 @@ public partial class EFXAttributePlayEfx : EFXAttribute
 	public EFXAttributePlayEfx() : base(EfxAttributeType.PlayEfx) { }
 
 	// [RszVersion(EfxVersion.MHWilds)] public uint mhws_unkn;
-	[RszInlineWString] public string? efxPath;
+	[RszInlineWString(ByteSize = true)] public string? efxPath;
 
     public override string ToString() => $"Efx path: {efxPath}";
 }
diff --git a/REE-Lib/OtherFiles/EFX/EfxEmitter.cs b/REE-Lib/OtherFiles/EFX/EfxEmitter.cs
index 68d8731..ca2566c 100644
--- a/REE-Lib/OtherFiles/EFX/EfxEmitter.cs
+++ b/REE-Lib/OtherFiles/EFX/EfxEmitter.cs
@@ -233,12 +233,12 @@ public partial class EFXAttributeMeshEmitterV2 : EFXAttribute
     public uint MaskVTilingCount;
     public Vector2 MaskUVOffset;
 
-	[RszInlineWString] public string? DynamicColorMapPath;
-	[RszInlineWString] public string? MeshPath;
-	[RszInlineWString] public string? MeshMirrorPath;
-	[RszInlineWString] public string? ColorMapPath;
-	[RszInlineWString] public string? MaskMapPath;
-	[RszInlineWString] public string? TargetGameObjectName;
+	[RszInlineWString(ByteSize = true)] public string? DynamicColorMapPath;
+	[RszInlineWString(ByteSize = true)] public string? MeshPath;
+	[RszInlineWString(ByteSize = true)] public string? MeshMirrorPath;
+	[RszInlineWString(ByteSize = true)] public string? ColorMapPath;
+	[RszInlineWString(ByteSize = true)] public string? MaskMapPath;
+	[RszInlineWString(ByteSize = true)] public string? TargetGameObjectName;
 
     public override string ToString() => !string.IsNullOrEmpty(MeshPath) ? MeshPath : type.ToString();
 }
diff --git a/REE-Lib/OtherFiles/EFX/EfxFluid.cs b/REE-Lib/OtherFiles/EFX/EfxFluid.cs
index e46abb6..337ece7 100644
--- a/REE-Lib/OtherFiles/EFX/EfxFluid.cs
+++ b/REE-Lib/OtherFiles/EFX/EfxFluid.cs
@@ -266,5 +272,5 @@ public partial class EFXAttributeFluidParticleEmitterTarget : EFXAttribute
 	public uint Unkn2;
 	public uint Unkn3;
 	public uint Unkn4;
-	[RszInlineWString] public string? UserDataFilePath;
+	[RszInlineWString(ByteSize = true)] public string? UserDataFilePath;
 }
diff --git a/REE-Lib/OtherFiles/EFX/EfxMiscStructs.cs b/REE-Lib/OtherFiles/EFX/EfxMiscStructs.cs
index 9d5cdf6..e154e94 100644
--- a/REE-Lib/OtherFiles/EFX/EfxMiscStructs.cs
+++ b/REE-Lib/OtherFiles/EFX/EfxMiscStructs.cs
@@ -519,7 +519,7 @@ public partial class EFXAttributeVanishArea3D : EFXAttribute, IBoneRelationAttri
     public Vector3 AreaScale;
     public Vector3 AreaAngle;
 	public via.Range VanishFrame;
-	[RszInlineWString] public string? JointName;
+	[RszInlineWString(ByteSize = true)] public string? JointName;
 
     public string? ParentBone { get; set; }
 
@@ -741,7 +741,7 @@ public partial class EFXAttributeFlowMap : EFXAttribute
     public via.Range Speed;
     public via.Range SpeedCoef;
 
-    [RszInlineWString] public string? flowmapMaskPath;
+    [RszInlineWString(ByteSize = true)] public string? flowmapMaskPath;
 
     public override string ToString() => !string.IsNullOrEmpty(flowmapMaskPath) ? flowmapMaskPath : type.ToString();
 }
@@ -754,10 +754,10 @@ public partial class EFXAttributeAssignCSV : EFXAttribute
     public bool RandomizeIndexOrder;
     public bool UseTableRange;
     public via.RangeI mTableRange;
-    [RszInlineWString] public string? efcsvPostionListPath;
-    [RszInlineWString] public string? efcsvRotationListPath;
-    [RszInlineWString] public string? efcsvVelocityListPath;
-    [RszInlineWString] public string? efcsvColorListPath;
+    [RszInlineWString(ByteSize = true)] public string? efcsvPostionListPath;
+    [RszInlineWString(ByteSize = true)] public string? efcsvRotationListPath;
+    [RszInlineWString(ByteSize = true)] public string? efcsvVelocityListPath;
+    [RszInlineWString(ByteSize = true)] public string? efcsvColorListPath;
 
     public override string ToString() => !string.IsNullOrEmpty(efcsvPostionListPath) ? efcsvPostionListPath : type.ToString();
 }
@@ -856,7 +856,7 @@ public partial class EFXAttributeLayout : EFXAttribute
 
 	// note: not 100% on the flag bits but most Wilds files seem to work with this
 	[RszConditional("((flags1 & (8 | 4 | 16)) != 0)", EndAt = nameof(layoutDataFloats))]
-	[RszInlineWString]
+	[RszInlineWString(ByteSize = true)]
 	public string? layoutName;
 	[RszFixedSizeArray]
 	public byte[]? layoutDataFloats;
diff --git a/REE-Lib/OtherFiles/EFX/EfxPtBehavior.cs b/REE-Lib/OtherFiles/EFX/EfxPtBehavior.cs
index 71f83ae..a7ac286 100644
--- a/REE-Lib/OtherFiles/EFX/EfxPtBehavior.cs
+++ b/REE-Lib/OtherFiles/EFX/EfxPtBehavior.cs
@@ -272,7 +272,7 @@ public partial class EFXAttributePtCollision : EFXAttribute
 	[RszConditional("(stringBitFlag & (1 << 0)) != 0"), RszInlineWString]
 	public string? unknString0;
 
-	[RszConditional("(stringBitFlag & (1 << 1)) != 0"), RszInlineWString]
+	[RszConditional("(stringBitFlag & (1 << 1)) != 0"), RszInlineWString(ByteSize = true)]
 	public string? unknString1;
 }
 
@@ -401,7 +401,7 @@ public partial class EFXAttributePtFreezer : EFXAttribute
 [RszGenerate, RszAutoReadWrite]
 public partial class PtPathTranslateName : BaseModel
 {
-	[RszInlineWString] public string name = "";
+	[RszInlineWString(ByteSize = true)] public string name = "";
 
     public override string ToString() => name;
 }
diff --git a/REE-Lib/OtherFiles/EFX/EfxTypeStrain.cs b/REE-Lib/OtherFiles/EFX/EfxTypeStrain.cs
index 7b2f0cf..5fe4886 100644
--- a/REE-Lib/OtherFiles/EFX/EfxTypeStrain.cs
+++ b/REE-Lib/OtherFiles/EFX/EfxTypeStrain.cs
@@ -228,7 +228,7 @@ public partial class EFXAttributeTypeStrainRibbonV3 : EFXAttribute
 	public float wilds_Ukn3_5;
 	public float wilds_Ukn3_6;
 	public UndeterminedFieldType wilds_UknLast;
-	[RszInlineWString] public string? boneName;
+	[RszInlineWString(ByteSize = true)] public string? boneName;
 
 	[RszVersion('<', EfxVersion.MHWilds, EndAt = nameof(unkn8_6))]
 	public int unkn8_0;
